### PR TITLE
Fix false positive in prefer_asserts_in_initializer_lists

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/linter/issues/1655

### CODE REVIEW

File: lib/src/rules/prefer_asserts_in_initializer_lists.dart [hunk 1]
```diff
@@ -101,8 +101,8 @@ class _AssertVisitor extends RecursiveAstVisitor {
   bool _hasMethod(MethodElement element) {
     final type = classElement.type;
     final name = element.name;
-    return type.lookUpMethod(name, element.library) == element ||
-        type.lookUpInheritedMethod(name) == element;
+    return element == type.lookUpMethod(name, element.library) ||
```
Comment (kevmoo) [inline lib/src/rules/prefer_asserts_in_initializer_lists.dart]: The `==` impl here is not symmetric, which is pretty annoying.

CC @scheglov

Comment (kevmoo) [reply lib/src/rules/prefer_asserts_in_initializer_lists.dart]: CC @bwilkerson

Comment (scheglov) [reply lib/src/rules/prefer_asserts_in_initializer_lists.dart]: Yeah, this stinks.

The reason why it is not symmetrical is that `type.lookUpMethod` might return `MethodMember`, which is `MethodElement` with type information (i.e. type parameters replaced with specific type arguments from the `type`). And `MethodMember` or `Member` at all does not implement `operator==`. It is not obvious what `==` means for members. Is it just comparison of base elements? Or also types of the member and the given element?

**Comment by kevmoo** [comment]
> I don't understand why "-dev" was removed, but you understand pub better than I do, so I'll trust that it's the right thing to do.

Some of us use `-dev` to signal the repo is not ready for publish.

But that breaks the `pana` test. It doesn't seem to be the flow on this repo.

**Comment by pq** [comment]
This is awesome. Thanks!

**Pull request merged**